src/Runner.Listener/MessageListener.cs

```diff
@@ -74,6 +92,25 @@ public async Task<Boolean> CreateSessionAsync(CancellationToken token)
             string errorMessage = string.Empty;
             bool encounteringError = false;
 
+            var v1Creds = _configStore.GetCredentials();
+            var v2Creds = _configStore.GetV2Credentials();
+            if (v1Creds != null)
+            {
+                _v1CredentialsExists = true;
```

Comment:
nit: looks like some unnecessary local vars adding extra complexity. It looks like v1creds, v2creds, and useLegacyAuthUrl are all that is needed. For instance, instead of managing _useV2Credentials, could set v2creds=null if need to fallback.

Reply:
nevermind now i see the motivation (2 day delay)

Reply:
perhaps only _v1CredentialsExists is unnecessary

Reply:
Another example, if env var GITHUB_ACTIONS_RUNNER_LEGACYAUTHURL  then could v2creds just be set to null. Would it simplify the logic to eliminate local variable `useLegacyAuthUrl`?

Reply:
`_v1CredentialsExists` is unnecessary, v1Cred should always exist.